### PR TITLE
[Merged by Bors] - feat(Subgraph): API lemmas

### PR DESCRIPTION
**Pull request by YaelDillies**
From LeanCamCombi

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Combinatorics/SimpleGraph/Subgraph.lean [hunk 1]
```diff
@@ -377,6 +394,18 @@ theorem verts_iSup {f : ι → G.Subgraph} : (⨆ i, f i).verts = ⋃ i, (f i).v
 @[simp]
 theorem verts_iInf {f : ι → G.Subgraph} : (⨅ i, f i).verts = ⋂ i, (f i).verts := by simp [iInf]
 
+@[simp] lemma coe_bot : (⊥ : G.Subgraph).coe = ⊥ := rfl
+
+@[simp] lemma IsInduced.top : (⊤ : G.Subgraph).IsInduced := fun _ _ ↦ id
+
+/-- The graph isomorphism between the top element of `G.subgraph` and `G`. -/
+def topIso : (⊤ : G.Subgraph).coe ≃g G where
```
Comment (b-mehta) [inline Mathlib/Combinatorics/SimpleGraph/Subgraph.lean:402]:
```suggestion
@[simps] def topIso : (⊤ : G.Subgraph).coe ≃g G where
```
maybe?

**Review by b-mehta — APPROVED** [review]
bors d+

File: Mathlib/Combinatorics/SimpleGraph/Subgraph.lean [hunk 2]
```diff
@@ -566,29 +599,26 @@ protected def map {G' : SimpleGraph W} (f : G →g G') (H : G.Subgraph) : G'.Sub
...
-      · exact Or.inl ⟨_, _, h, rfl, rfl⟩
-      · exact Or.inr ⟨_, _, h, rfl, rfl⟩
-    · rintro (⟨a, b, h, rfl, rfl⟩ | ⟨a, b, h, rfl, rfl⟩)
-      · exact ⟨_, _, Or.inl h, rfl, rfl⟩
-      · exact ⟨_, _, Or.inr h, rfl, rfl⟩
+    exact ⟨_, _, hH.2 ha, rfl, rfl⟩
+
+lemma map_monotone : Monotone (Subgraph.map f) := fun _ _ ↦ map_mono
+
+theorem map_sup (f : G →g G') (H₁ H₂ : G.Subgraph) : (H₁ ⊔ H₂).map f = H₁.map f ⊔ H₂.map f := by
```
[the remark this answers is not in the document]
Comment (YaelDillies) [reply Mathlib/Combinatorics/SimpleGraph/Subgraph.lean:617]: `map_inf` is not true: Consider the map `G ⊕ G →g G` (where `⊕` is abuse of notation for two disjoint copies of `G`) collapsing the two copies of `G`. Then `map (left G inf right G) = map bot = bot` but `map (left G) inf map (right G) = G inf G = G`

File: Mathlib/Combinatorics/SimpleGraph/Subgraph.lean (hunk 1 above)
Comment (YaelDillies) [reply Mathlib/Combinatorics/SimpleGraph/Subgraph.lean:402]: Already tried and failed 😁

File: Mathlib/Combinatorics/SimpleGraph/Subgraph.lean (hunk 2 above)
[the remark this answers is not in the document]
Comment (YaelDillies) [reply Mathlib/Combinatorics/SimpleGraph/Subgraph.lean:617]: > More generally, don't these both follow from some more general order-theory thing?

I suspect there is a galois connection with `comap` but I haven't worked out the details

**Pull request closed without merging**